### PR TITLE
[core] Move UsageStatsClient to gcs_server/.

### PR DESCRIPTION
**Pull request by rynewang**
Previously GcsServer keeps a GcsClient to itself. This is awkward and require careful placement of order of init calls. The dependency is not needed: the only user is UsageStatsClient which is only used by GcsServer, so we can move it into gcs_server and let it directly talk to InternalKV Manager.

### CODE REVIEW

File: src/ray/gcs/gcs_server/usage_stats_client.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 The Ray Authors.
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/usage_stats_client.cc:1]: maybe 2024? lol

File: src/ray/gcs/gcs_server/test/usage_stats_client_test.cc [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Ray Authors.
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/test/usage_stats_client_test.cc:1]: 2024

**Review by rkooo567 — APPROVED** [review]
great changes;:)

**Comment by jjyao** [comment]
I think we planned to use it inside raylet etc as well.

**Review by jjyao — APPROVED** [review]

**Comment by rynewang** [comment]
There was a linux cpp test failure due to order of init (now InitUsageStatsClient needs to happen before rpc server start). Updated and merging...

**Pull request merged**